server/server.go

```diff
@@ -142,6 +193,14 @@ func (s *RpcEndPointServer) HandleHttpRequest(respw http.ResponseWriter, req *ht
 	request.process()
 }
 
+func (s *RpcEndPointServer) handleDrain(respw http.ResponseWriter, req *http.Request) {
+	if s.isHealthy {
+		s.isHealthy = false
+		s.logger.Info("Server marked as unhealthy")
+		time.Sleep(60 * time.Second) // Give LB enough time to detect us unhealthy
```

Comment:
So the idea is that request to the drain endpoint should take 60 seconds?

Reply:
yes.

reason is that control plane considers the pre-hook to be "done" right at the moment when the request returns a status, which is when it will send the SIGTERM to the pod.